src/crypto/keys/common_attrs.rs

```diff
@@ -0,0 +1,22 @@
+use crate::bip32::child_number::ChildNumber;
+
+const FINGERPRINT_LEN: usize = 4;
+
+pub type KeyFingerprint = [u8; FINGERPRINT_LEN];
+
+#[derive(Clone, Copy)]
+pub struct ExtendedKeyAttrs {
+    pub depth: u8,
+    pub parent_fingerprint: KeyFingerprint,
+    pub child_number: ChildNumber,
+}
+
+impl ExtendedKeyAttrs {
+    pub fn new(depth: u8, parent_fingerprint: KeyFingerprint, child_number: ChildNumber) -> Self {
+        Self { depth: depth, parent_fingerprint: parent_fingerprint, child_number: child_number }
```

Comment:
Maybe we can directly set the index (just the unsigned integer) so that we don't have to worry about is_hardened here.